manifests/system.pp

```diff
@@ -67,11 +56,13 @@
 
   }
   else {
+    ensure_packages(['curl'])
```

Comment:
Do you have Windows in mind? In which case, the `path`, the `command` and the `creates` in the following resource would also be wrong.

And even then, even if it was still relevant, there is still no oint in checking whether or not it's been defined first, the goal of `ensure_packages` is that you can call it even if the package hasn't been defined first.